packages/docusaurus-plugin-content-docs/src/sidebars/utils.ts

```diff
@@ -266,6 +276,45 @@ Available document ids are:
     }
   }
 
+  function getFirstDocOfSidebar(sidebar: Sidebar):
```

Comment:
What about refactor to following signature?

```ts
function findFirstLink(sidebar: Sidebar): {label: string, path: string} | undefined
```

Reply:
So to make this function return the doc's path, I passed into it a `LoadedVersion` object to look up the doc's permalink:

```ts
function findFirstLink(sidebar: Sidebar, version: LoadedVersion): { label: string; path: string } | undefined
```

I just refactored the function in my recent commit. Let me know if you think this is a good approach, or if you think it's a bit overkill to pass `LoadedVersion` inside this function.

Reply:
I think it's overkill. Passing in such a big entity makes this function very hard to reason about and optimize in the future. I prefer the initial implementation: the return type is a bit complicated, but it's lightweight and easy to be piped into other processing.